opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -132,6 +132,16 @@ def add_link(self: 'Span',
         `SpanContext` passed as argument.
         """
 
+    def update_name(self: 'Span', name: str) -> None:
+        """Updates the `Span` name.
+
+        This will override the name provided via :func:`Tracer.create_span`
+        or :func:`Tracer.start_span`.
+
+        Upon this update, any sampling behavior based on Span name will depend
+        on the implementation.
+        """
+
```

Comment:
Shouldn't it raise a `NotImplementedError`?

```suggestion
        raise NotImplementedError

```

Reply:
My thinking was (and I thought we decided that somewhere, but can't find it) that since the SDK package must be usable on its own as a no-op implementation, we would make the SDK interface the no-op implementation instead of raising exceptions (that would require a separate NoOp version of each class in the SDK which is a lot of boilerplate code).

Reply:
➡️  #66